Libraries/LibWeb/CookieStore/CookieStore.h

```diff
@@ -6,19 +6,41 @@
 
 #pragma once
 
+#include <AK/Optional.h>
+#include <AK/String.h>
+#include <LibWeb/Bindings/CookieStorePrototype.h>
 #include <LibWeb/DOM/EventTarget.h>
+#include <LibWeb/HighResolutionTime/DOMHighResTimeStamp.h>
 
 namespace Web::CookieStore {
 
+// https://cookiestore.spec.whatwg.org/#dictdef-cookielistitem
+struct CookieListItem {
+    Optional<String> name;
+    Optional<String> value;
+};
+
 // https://cookiestore.spec.whatwg.org/#cookiestore
 class CookieStore final : public DOM::EventTarget {
     WEB_PLATFORM_OBJECT(CookieStore, DOM::EventTarget);
     GC_DECLARE_ALLOCATOR(CookieStore);
 
+public:
+    GC::Ref<WebIDL::Promise> get(String name);
+
 private:
-    CookieStore(JS::Realm&);
+    CookieStore(JS::Realm&, PageClient&);
 
     virtual void initialize(JS::Realm&) override;
+    virtual void visit_edges(Cell::Visitor&) override;
+
+    GC::Ref<PageClient> m_client;
 };
 
 }
+
+namespace Web::Bindings {
+
+JS::Value cookie_list_item_to_value(JS::Realm&, CookieStore::CookieListItem);
```

Comment:
Hmm can the declaration be generated in the prototype header?

Reply:
no, I tried doing that initially, but since the dictionary struct is defined in the non-generated header, we would have to include that header in the prototype header, and that would create an include loop (since the non-generated header also needs to include the prototype header)

Reply:
Oh wait, just noticed - the generated method should be taking the value to be converted as a `const&` anyways. Then a forward declaration of `CookieListItem` can be added to `LibWeb/Forward.h`, or generated - then this method declaration can also be generated.

Reply:
I checked, this is still non-trivial, since we don't have a solution at the moment in the IDL generator to resolve the namespace of things - i.e. in the cpp files we just add `using namespace` for every single namespace,  which we wouldn't want to do in header files.
If we still want to do this, we would have to declare the dictionary in the global Web namespace

Reply:
Got it, let's not for now then!
We should still take the param by `const&` though, no need to copy